artiq/dashboard/datasets.py

```diff
@@ -134,14 +134,15 @@ def accept(self):
     def dtype(self):
         txt = self.value_widget.text()
         try:
-            self.data_type.setText(type(pyon.decode(txt)).__name__)
-            self.value_widget.setStyleSheet("")
-            self.ok.setEnabled(True)
+            pyon.decode(txt)
```

Comment:
Just store the result and decode once. You don't need to do anything special regarding try/else:
```python
>>> try:
...    x = 2
... except:
...    pass
... else:
...    print(x)
... 
2
```

Please do such research on the Python language next time.